dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/engine/graph/WorkflowExecutionGraph.java

```diff
@@ -335,7 +336,7 @@ public boolean isAllSuccessorsAreConditionTask(final ITaskExecutionRunnable task
         }
         return successors.stream().allMatch(
                 successor -> isTaskExecutionRunnableSkipped(successor)
-                        || TaskTypeUtils.isConditionTask(taskExecutionRunnable.getTaskInstance().getTaskType()));
+                        || TaskTypeUtils.isConditionTask(successor.getTaskDefinition().getTaskType()));
```

Comment:
We may need to filter the forbidden susseccors here, 
```java
(TaskTypeUtils.isConditionTask(successor.getTaskDefinition().getTaskType()) && !isTaskExecutionRunnableForbidden(successor))
```
if the successor condition task is forbidden, then should return false.